staging/src/k8s.io/client-go/tools/watch/retrywatcher.go

```
@@ -126,6 +126,22 @@ func (rw *RetryWatcher) doReceive() (bool, time.Duration) {
 			return false, 0
 		}
 
+		// Check if the watch failed due to the client not having permission to watch the resource or the credentials
+		// being invalid (e.g. expired token).
+		if apierrors.IsForbidden(err) || apierrors.IsUnauthorized(err) {
+			// Add more detail since the forbidden message returned by the Kubernetes API is just "unknown".
+			klog.ErrorS(err, msg+": ensure the client has watch permissions on the resource")
+
+			statusErr := err.(apierrors.APIStatus).Status()
+
+			_ = rw.send(watch.Event{
```

don't ignore the error